source/building-your-own-javascript-components/index.html.md.erb

```diff
@@ -0,0 +1,283 @@
+// If the component does not need any configuration, pass `null` for the configuration
+createAll(ProjectComponent, null, $element)
+```
+
+### Handling initialisation errors
+
+By default, `createAll` catches errors thrown by components during their initialisation and logs them in the console. This makes sure components further down the page still get initialised after an error. You may still want to respond to errors happening during component initialisation (for example, notifying an error monitoring service) without [manually initialising each component](../import-javascript/#import-individual-components). 
+
+You can use a function as a third argument to respond to errors being thrown while components are being initialised. If a component throws an error, the function will be called and will receive:
+
+- the error that was thrown
+- an object with some more context
+
+The context object will contain the following properties:
+
+- `Component`: The component class being initialised
+- `element`: The element the component is being initialised on
+- `config`: The configuration used for initialisation
+
+```js
+import {createAll} from ‘govuk-frontend’
+import {ProjectComponent} from ‘./project-component.js’
+
+function notifyErrorMonitoringService(error, { Component, element, config }) {
+  // Send relevant details to an error monitoring service
+}
+
+createAll(ProjectComponent, {SOME_CONFIGURATION}, notifyErrorMonitoringService)
+// If you don’t need any configuration, pass `null` for the configuration
+createAll(ProjectComponent, null, notifyErrorMonitoringService)
+```
+
+If you need to [initialise only on part of the page](#initialise-only-on-part-of-a-page) and respond to errors, you can pass an object in that third argument with the following properties:
+
+`scope`: The element within which to look for components to initialise
+`onError`: The callback for responding to errors
+
+```js
+import {createAll} from ‘govuk-frontend’
+import {ProjectComponent} from ‘./project-component.js’
+
+const $element = document.querySelector(‘.app-dialog’)
+
+function notifyErrorMonitoringService(error, { Component, element, config }) {
+  // Do something with initialisation error, like sending relevant details to an error monitoring service
+}
+
+createAll(ProjectComponent, {SOME_CONFIGURATION}, {
+  scope: $element,
+  onError: notifyErrorMonitoringService
+})
+// If you don’t need any configuration, pass `null` for the configuration
+createAll(ProjectComponent, null, {
+  scope: $element,
+  onError: notifyErrorMonitoringService
+})
+```
+
+## Checking for GOV.UK Frontend support with `isSupported()`
+
+GOV.UK Frontend components and components inheriting from [our `Component` class](#using-the-component-class) will automatically check if GOV.UK Frontend is supported during their initialisation. However, you may want to check support separately, to avoid running code unnecessarily if GOV.UK Frontend is not supported.
```

Comment:
Similar to Line 27 – this gerund ('-ing') seems a bit confusing, this suggestion might read better if it's still accurate.

```suggestion
GOV.UK Frontend components and components that inherit from [our `Component` class](#using-the-component-class) will automatically check if GOV.UK Frontend is supported during their initialisation. However, you may want to check support separately, to avoid running code unnecessarily if GOV.UK Frontend is not supported.
```

Reply:
The change of the `-ing` form reads OK, but the edit on the end of the last sentence changes the meaning. The idea is that services may want to run their own code alongside GOV.UK Frontend, but only run it if GOV.UK Frontend is supported. This could be because their code depends on our components initialising before they react to events on the elements inside the component that are only present if GOV.UK Frontend is supported.

I'll keep the ending `if` clause on that last statement, but take the rest of the edits:

```suggestion
GOV.UK Frontend components and components that inherit from [our `Component` class](#using-the-component-class) will automatically check if GOV.UK Frontend is supported during their initialisation. However, you may want to separately check for support, to avoid unnecessarily running code if GOV.UK Frontend is not supported.
```

Reply:
@calvin-lau-sig7 Kept this one as unresolved given I edited your edit, let me know if it's OK 😊